### DISCUSSION TITLE
Linked transfers and expenses

### DISCUSSION THREAD

**Discussion by Jademalo** [firefly-iii/firefly-iii#7672] [Firefly III]
I'm curious if it's possible to do this, as it makes PayPal account management a lot easier.

I do a large amount of my transactions via PayPal. When I do these transactions, it first takes from my PayPal balance, and then takes from my Credit Card.

In my previous app, I had two accounts - Credit Card, and PayPal. 
When I would buy anything via PayPal, I would first make an expense in there, and then make a transfer between it and my Credit Card. This meant that if there was balance in the PayPal account I would have the right amount debited from the Credit Card and everything would balance nicely.
The alternative was two transactions for the same item, one for the PayPal balance part and one for the Credit Card part. I wasn't a huge fan of this though since I couldn't easily see all of the transactions that went through PayPal.

In Firefly, Is there any way to create an expense that has a transfer component to it?
So, for example, I make an expense of £20 in PayPal, and it then automatically creates an associated transfer for the £20 from my Credit Card.

I had a look at rules, but I couldn't figure out a way to duplicate a transaction as a transfer using them.

Thanks!

**Comment by JC5** [comment]
You can link transactions, but you can't combine splits this way I'm afraid.

**Reply by Jademalo** [reply]
Damn, that's a shame! Thanks for the quick reply.

I can't really think of a better way of doing it other than manually as I am now, unless I only show PayPal transactions that only use my card in my card's account. Linked transactions are ok for those that split the balance, but for those that don't it obviously doesn't show up in both places.

For the sake of convenience, is there any way to generate new transactions via rules, or are they purely for modifying existing transactions?
Failing that, is there a way to have a linked transaction with a "£0.00" component so it at least correctly shows up in the PayPal list?

Thanks again!

**Reply by JC5** [reply]
> For the sake of convenience, is there any way to generate new transactions via rules, or are they purely for modifying existing transactions?

You can use webhooks, but you'll have to develop something yourself. The rules can delete transactions, but they can't create it.

>  Failing that, is there a way to have a linked transaction with a "£0.00" component so it at least correctly shows up in the PayPal list?

No, there has to be an amount present.

**Reply by Jademalo** [reply]
Alright, guess I'll just keep using my current method then.
Thanks for the quick replies, much appreciated!
